### PR TITLE
Move to 3.0 Preview 6 SDK

### PR DESCRIPTION
**Pull request by dougbu**

### CODE REVIEW

**Review by JunTaoLuo — APPROVED** [review]

**Review by JunTaoLuo — CHANGES_REQUESTED** [review]
Approved too early. Looks like migrating to the new SDK broke parts of our build.

**Comment by anurse** [comment]
> Looks like migrating to the new SDK broke parts of our build.

This is the pain induced by moving so late :(.

**Comment by Pilchie** [comment]
Tagging to see this.

**Comment by JunTaoLuo** [comment]
From email, it seems like we need a newer version of VS. I've switched to using the VS preview queues. We'll see whether the tests feel better after this.

**Comment by JunTaoLuo** [comment]
Okay that didn't seem to work. We'll apply the workaround instead.

File: eng/targets/ReferenceAssembly.targets [hunk 1]
```diff
@@ -76,10 +76,6 @@
     </ItemGroup>
 
     <PropertyGroup>
-      <ProjectListFile>$(MSBuildThisFileDirectory)..\eng\ProjectReferences.props</ProjectListFile>
-      <RefsTargetFramework>$(TargetFrameworks)</RefsTargetFramework>
-      <RefsTargetFramework Condition="'$(RefsTargetFramework)' == ''">$(TargetFramework)</RefsTargetFramework>
```
Comment (dougbu) [inline eng/targets/ReferenceAssembly.targets]: Minor nit: These properties are unused. Recreated ref assemblies are unchanged after removing them.

**Comment by dougbu** [comment]
:up::date: to add dotnet/cli#11378 workaround (NuGet versions in current use do not honour `PrivateAssets="All"` in `@(FrameworkReference)` items). Local builds are now clean.

**Comment by JunTaoLuo** [comment]
Thanks. Code check failure looks like flakiness. Not sure about the site extension error though. I'll re-run it and see if it's flakiness or something is really broken.

**Comment by JunTaoLuo** [comment]
Well it seems like the SDK actually broke SiteExtensions build. Investigating.

**Comment by dougbu** [comment]
Thanks @JunTaoLuo❕

File: src/SiteExtensions/build.cmd [hunk 2]
```diff
@@ -10,8 +10,9 @@ IF %ERRORLEVEL% NEQ 0 (
 )
 
 ECHO Building LoggingBranch
-CALL %RepoRoot%\build.cmd -forceCoreMsbuild -arch x64 -projects %~dp0LoggingBranch\LB.csproj /bl:artifacts/log/SiteExtensions-LoggingBranch-x64.binlog %*
-CALL %RepoRoot%\build.cmd -forceCoreMsbuild -arch x86 -projects %~dp0LoggingBranch\LB.csproj /bl:artifacts/log/SiteExtensions-LoggingBranch-x86.binlog %*
+REM Remove /p:DisablePackageAssetsCache=true /p:DisableTransitiveFrameworkReferences=true which workarounds dotnet store and publish behaviour changes in preview 6
```
Comment (dougbu) [inline src/SiteExtensions/build.cmd]: This comment isn't clear. Is it a TODO for when we move to Preview 7 perhaps? Is there a bug tracking a fix in the SDK that should be linked here?

**Comment by dougbu** [comment]
Thanks very much @JunTaoLuo❕

File: src/SiteExtensions/build.cmd (hunk 2 above)
Comment (JunTaoLuo) [reply src/SiteExtensions/build.cmd]: I think we need a little more clarification on this matter, which I'll follow up with tomorrow. But to potentially unblock us, I wanted to test this change on the CI (works locally now). I'll send an email with a few more details to @mkArtakMSFT once I get a build pass/fail.

**Comment by dougbu** [comment]
@JunTaoLuo do you approve this PR now?

**Review by JunTaoLuo — APPROVED** [review]
I want to see the build succeed first though.

**Comment by dougbu** [comment]
> I want to see the build succeed first though.

Me too 😈

**Comment by JunTaoLuo** [comment]
It's at the artifact publishing step now so I think it's good to merge. I'll do some additional verification of the built site extensions tomorrow.

**Pull request merged**